code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -503,53 +503,63 @@
-				movement_target = null
-				stop_automated_movement = 0
-				for(var/obj/item/reagent_containers/food/snacks/S in oview(src,3))
-					if(isturf(S.loc) || ishuman(S.loc))
-						movement_target = S
+	if(resting || buckled)
+		return
+
+	if(++turns_since_scan > 5)
+		turns_since_scan = 0
+
+		// Has a target, but it's not where it was before, and it wasn't picked up by someone.
+		if(movement_target && !(isturf(movement_target.loc) || ishuman(movement_target.loc)))
+			movement_target = null
+			stop_automated_movement = FALSE
+		// No current target, or current target is out of range.
+		if(!movement_target || !(movement_target.loc in oview(src, 3)))
+			movement_target = null
+			stop_automated_movement = FALSE
+			var/obj/item/possible_target = null
+			for(var/I in oview(src, 3))
+				if(istype(I, /obj/item/reagent_containers/food/snacks)) // Noms
+					possible_target = I
+					break
+				else if(istype(I, /obj/item/paper)) // Important noms
+					if(prob(10))
+						possible_target = I
 						break
-			if(movement_target)
-				spawn(0)
-					stop_automated_movement = 1
-					step_to(src,movement_target,1)
-					sleep(3)
-					step_to(src,movement_target,1)
-					sleep(3)
-					step_to(src,movement_target,1)
-
-					if(movement_target)		//Not redundant due to sleeps, Item can be gone in 6 decisecomds
-						if(movement_target.loc.x < src.x)
-							dir = WEST
-						else if(movement_target.loc.x > src.x)
-							dir = EAST
-						else if(movement_target.loc.y < src.y)
-							dir = SOUTH
-						else if(movement_target.loc.y > src.y)
-							dir = NORTH
-						else
-							dir = SOUTH
-
-						if(!Adjacent(movement_target)) //can't reach food through windows.
-							return
-
-						if(isturf(movement_target.loc) )
-							movement_target.attack_animal(src)
-						else if(ishuman(movement_target.loc) )
-							if(prob(20))
-								custom_emote(1, "stares at [movement_target.loc]'s [movement_target] with a sad puppy-face")
+			if(possible_target && (isturf(possible_target.loc) || ishuman(possible_target.loc))) // On the ground or in someone's hand.
+				movement_target = possible_target
+		if(movement_target)
+			INVOKE_ASYNC(src, .proc/move_to_target)
+
+	if(prob(1))
+		chasetail()
+
+/mob/living/simple_animal/pet/dog/corgi/Ian/proc/move_to_target()
+	stop_automated_movement = TRUE
+	step_to(src, movement_target, 1)
+	sleep(3)
+	step_to(src, movement_target, 1)
+	sleep(3)
+	step_to(src, movement_target, 1)
+
+	if(movement_target) // Not redundant due to sleeps, Item can be gone in 6 deciseconds
+		// Face towards the thing
+		if(movement_target.loc.x < src.x)
+			dir = WEST
+		else if(movement_target.loc.x > src.x)
+			dir = EAST
+		else if(movement_target.loc.y > src.y)
+			dir = NORTH
+		else // Face south if you're on or above the tile
+			dir = SOUTH
+
+		if(!Adjacent(movement_target)) //can't reach food through windows.
+			return
 
-		if(prob(1))
-			custom_emote(1, pick("dances around.","chases its tail!"))
-			spin(20, 1)
```

Comment:
Why was this removed?

Reply:
It was replaced with
```dm
	if(prob(1))
		chasetail()
```
Since that verb does more or less exactly the same thing.
```dm
/mob/living/simple_animal/pet/dog/verb/chasetail()
	set name = "Chase your tail"
	set desc = "d'awwww."
	set category = "Dog"

	visible_message("[src] [pick("dances around", "chases [p_their()] tail")].", "[pick("You dance around", "You chase your tail")].")
	spin(20, 1)
```

Reply:
But isn't the code I mentioned part of the dog's AI instead of being a player-facing verb?

Reply:
It is, yes. I changed the AI code to call the `chasetail()` verb rather than `custom_emote()` and `spin()`, since what was there was pretty much 1:1 with the verb.